src/components/Element/Article.jsx

```diff
@@ -0,0 +1,97 @@
+import clsx from 'clsx';
+import { useState, useEffect, useRef } from 'react';
+import { Tip, Content, useArticleContext } from './ArticleElement';
+import { maxLength } from '../../scripts';
+
+export function Article() {
+  const { FetchData, message } = useArticleContext();
+  const [article, setArticle] = useState([]);
+  const [sentence, setSentence] = useState({});
+  const [position, setPosition] = useState({});
+  const [link, setLink] = useState('');
+  const markRef = useRef(null);
+
+  function getTipPosition() {
+    const { scrollY } = window;
+    setPosition({
+      left: markRef.current.offsetLeft,
+      top: markRef.current.offsetTop - scrollY,
+    });
+  }
+
+  function showTip(e) {
+    e.preventDefault();
+    const selection = window.getSelection();
+    const range = selection.getRangeAt(0);
+    const start = range.startOffset;
+    const end = range.endOffset;
+    const text = article.content;
+    const word = text.substring(start, end);
+    const dist = {
+      prev: text.substring(0, start),
+      word: word,
+      next: text.substring(end, text.length),
+    };
+    word.trim() && setSentence(dist);
+  }
+
+  function hideTip(e) {
+    !e.target.closest('.Tip') && setSentence({});
+  }
+
+  function translateURL() {
+    if (sentence.word) {
+      const url = 'https://translate.google.com.tw';
+      const search = {
+        hl: 'zh-TW',
+        tab: 'rT',
+        sl: 'en',
+        tl: 'zh-TW',
+        text: sentence.word.trim(),
+        op: 'translate',
+      };
+      const result = `${url}/?` + new URLSearchParams({ ...search });
+      return result.toString().split('+').join('%20');
+    }
+  }
+
+  async function fetchArticle() {
+    const data = await FetchData(`${process.env.REACT_APP_URL}/api/article`, 'get');
```

Comment:
destructor

Reply:
data will be null if there's an error. Should try and avoid multiple types for  states